### PR TITLE
Optional CS config

### PR DESCRIPTION
**Pull request by gpupo**
- Add CS config;
- Declare strict_types; 
- Add File Header; 
- apply CS rules

### CODE REVIEW

**Comment by gpupo** [comment]
Strict mode broke the tests because shows some points to enforce the types and uses

File: .php_cs.dist [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?php 
+
+use Gpupo\CommonDevExtra\CsConfigurator;
```
Comment (mghoneimy) [inline .php_cs.dist:3]: Where is this package required? And what does it do?

File: .gitignore [hunk 2]
```diff
@@ -4,3 +4,4 @@ local_*/
 .phpunit.result.cache
 /build/
 composer.lock
+.*.cache
```
Comment (mghoneimy) [inline .gitignore:7]: What generates this?

**Review by mghoneimy — COMMENTED** [review]
What CS standard are you using here? I don't understand why you are replacing `assertEquals` with `assertSame` everywhere

File: .php_cs.dist (hunk 1 above)
Comment (gpupo) [reply .php_cs.dist:3]: this is required at composer global env and that set the php-cs-fixer config. If the project adopt that changes, we can replace with pure php-cs-fixer config

File: .gitignore (hunk 2 above)
Comment (gpupo) [reply .gitignore:7]: php-cs-fixer

File: .php_cs.dist (hunk 1 above)
Comment (gpupo) [reply .php_cs.dist:3]: assertEquals does not take datatype into consideration and type is one big focus of new level of php :)

Comment (gpupo) [reply .php_cs.dist:3]: @mghoneimy i propose to make a release with #46 and after, change the composer.json to php8 only, to give to the project things like named parameters and better types. A new major version, so then with this PR

**Pull request closed without merging**